src/shared/defaultAwsContext.ts

```diff
@@ -89,7 +92,29 @@ export class DefaultAwsContext implements AwsContext {
             await this._credentialsMru.setMostRecentlyUsedProfile(this.profileName)
         }
 
-        this._onDidChangeContext.fire(new ContextChangeEventsArgs(this.profileName, this.explorerRegions))
+        this._onDidChangeContext.fire(new ContextChangeEventsArgs(
+            this.profileName,
+            this.accountId,
+            this.explorerRegions
+        ))
+    }
+
+    // returns the configured profile's account ID, if any
+    public getCredentialAccountId(): string | undefined {
+        return this.accountId
+    }
+
+    // resets the context to the indicated profile, saving it into settings
+    public async setCredentialAccountId(accountId?: string): Promise<void> {
+        this.accountId = accountId
+        await this.settingsConfiguration.writeSetting(accountIdSettingKey, accountId, vscode.ConfigurationTarget.Global)
```

Comment:
I agree with the first option--this will ensure that at the very least the initial credentials line up.

We do need to also take https://github.com/aws/aws-toolkit-vscode/issues/168 into account...right now, we don't account for manual changes to the credentials file, but at least this gets us off on the right foot. For now, this issue is out of scope, but the account-on-extension-startup remains in.